apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -134,30 +153,46 @@ export class ProjectBuildCache {
       )
     );
 
-    const tarStream: stream.Writable = tar.extract({ cwd: projectFolderPath });
-    const extractTarPromise: Promise<boolean> = new Promise(
-      (resolve: (result: boolean) => void, reject: (error: Error) => void) => {
-        try {
-          tarStream.on('error', (error: Error) => reject(error));
-          tarStream.on('close', () => resolve(true));
-          tarStream.on('drain', () => resolve(true));
-          tarStream.write(cacheEntryBuffer);
-        } catch (e) {
-          reject(e);
-        }
+    const tarExecutablePath: string | undefined = ProjectBuildCache._tarExecutablePath;
+    let restoreSuccess: boolean;
+    if (!tarExecutablePath || !localCacheEntryPath) {
+      if (!cacheEntryBuffer && localCacheEntryPath) {
+        // eslint-disable-next-line require-atomic-updates
+        cacheEntryBuffer = await FileSystem.readFileToBufferAsync(localCacheEntryPath);
```

Comment:
Why don't we just try/catch this in the first place, instead of doing a separate file existence check? It'd be faster, unless our file existence check is coming from a cached readdir call.

Reply:
We don't need to read the file into memory if the native tar works, but we do need to know if the file exists before we invoke tar on it.